src/API/AbstractAPI.php

```diff
@@ -70,11 +72,22 @@ protected function get(string $path, array $query = [])
      */
     protected function post(string $path, array $parameters = [])
     {
-        $response = $this->connection->getHttpClient()->post(
-            $path,
+        $response = $this->client->getHttpClient()->post(
+            $this->buildUrl('post', $path),
             ['json' => $parameters]
         );
 
         return json_decode($response->getBody()->getContents(), true);
     }
+
+    private function buildUrl(string $method, string $path): string
+    {
+        if ($method === 'post') {
+            $baseUri = Arr::get($this->client->getHosts(), 'transactions', $this->client->getHosts()['api']);
+        } else {
+            $baseUri = $this->client->getHosts()['api'];
+        }
```

Comment:
🤔 this works for now but makes it hard to extend. Can we pass it a property that defines which API type it will use? by default it will then use the regular api, but the methods can pass a property to override it for the tx/evm api endpoints instead?

Reply:
@ItsANameToo added a method to override a single host [here](https://github.com/ArkEcosystem/php-client/pull/111/commits/a8237d67ccc7edd5db7c2867d1cb8f9533016c8b), let me know what do you think